keps/sig-node/0000-apparmor-ga/README.md

```diff
@@ -0,0 +1,685 @@
+reduce breakage. If this feature is implemented in v1.20, I propose v1.24 as a
+target for removal of the old behavior.
+
+### Test Plan
+
+AppArmor already has [e2e tests][https://github.com/kubernetes/kubernetes/blob/6596a14/test/e2e_node/apparmor_test.go],
+but the tests are guarded by the `[Feature:AppArmor]` tag and not run in the
+standard test suites.
+
+Prior to being marked GA, the feature tag will be removed from the AppArmor
+tests, and the tests will be migrated to the new fields API. Tests will be
+tagged as `[LinuxOnly]`.
+
+New tests will be added covering the annotation/field conflict cases described
+under [Version Skew Strategy](#version-skew-strategy).
+
+Test coverage for localhost profiles will be added, unless we decide to [keep
+localhost support in alpha](#alternatives).
+
+### Graduation Criteria
+
+_This section is excluded, as it is the subject of the entire proposal._
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable, can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md
+
+Production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+- **How can this feature be enabled / disabled in a live cluster?**
+
+  This feature was implemented prior to the concept of feature gates. Meaning,
+  it is already enabled in live clusters and cannot be disabled. This KEP is
+  simply "cleaning up" the API to make it GA.
+
+- **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+
+  This feature (AppArmor) cannot be disabled, as it is currently in a "pseudo-GA"
+  state. However, the changes it brings are backwards compatible, and the API
+  supports rollback of the kubernetes apiserver as described in the [Version
+  Skew Strategy](#version-skew-strategy).
+
+- **What happens if we reenable the feature if it was previously rolled back?**
+
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
```

Comment:
Earlier in this document it mentions app armor was added in v1.4, which seems to align with [the docs](https://kubernetes.io/docs/tutorials/clusters/apparmor/). Not sure if maybe it was alpha for one release before becoming beta? Or should this also read 1.4?

Reply:
Yep, I think 1.4 is correct. :+1: Fixed.